exporter/loggingexporter/factory.go

```diff
@@ -43,28 +44,54 @@ func (f *Factory) CreateDefaultConfig() configmodels.Exporter {
 			TypeVal: typeStr,
 			NameVal: typeStr,
 		},
+		LogLevel: "info",
 	}
 }
 
-func noopStopFunc() error {
-	return nil
-}
-
 // CreateTraceExporter creates a trace exporter based on this config.
-func (f *Factory) CreateTraceExporter(logger *zap.Logger, cfg configmodels.Exporter) (consumer.TraceConsumer, exporter.StopFunc, error) {
+func (f *Factory) CreateTraceExporter(logger *zap.Logger, config configmodels.Exporter) (consumer.TraceConsumer, exporter.StopFunc, error) {
+	cfg := config.(*Config)
 
-	lexp, err := NewTraceExporter(cfg.Name(), logger)
+	le, err := f.createLoggingExporter(logger, cfg.LogLevel)
 	if err != nil {
 		return nil, nil, err
 	}
-	return lexp, noopStopFunc, nil
+
+	lexp, err := NewTraceExporter(cfg.Name(), le.logger)
+	if err != nil {
+		return nil, nil, err
+	}
+	return lexp, le.stop, nil
+}
+
+func (f *Factory) createLoggingExporter(logger *zap.Logger, logLevel string) (*loggingExporter, error) {
+	var level zapcore.Level
+	err := (&level).UnmarshalText([]byte(logLevel))
+	if err != nil {
+		return nil, err
+	}
+	conf := zap.NewProductionConfig()
+	conf.Level.SetLevel(level)
```

Comment:
I'm less familiar with zap logger, do you need to update this line as well: https://github.com/open-telemetry/opentelemetry-service/pull/257/files#diff-f20f48383dee213429a81b242322b371R37?

Reply:
Do you mean to change from `logger.Debug(` to `logger.Info(`, which is the default level in the logging exporter?

Reply:
https://github.com/open-telemetry/opentelemetry-service/pull/257/files#diff-f20f48383dee213429a81b242322b371R37 and https://github.com/open-telemetry/opentelemetry-service/pull/257/files#diff-f20f48383dee213429a81b242322b371L47 are the only outputs of logger. If the exporter always calls `logger.Debug` or `logger.Info`, IIUC setting a `logLevel` won't be useful.

Reply:
Ah, I see what you meant. Actually that logger instance is the one created by https://github.com/open-telemetry/opentelemetry-service/pull/257/files#diff-91dcfeb5a40c1f4d3b45353b157e0bb4R87 and passed to https://github.com/open-telemetry/opentelemetry-service/pull/257/files#diff-91dcfeb5a40c1f4d3b45353b157e0bb4R92, which is different than the logger from `exporter_builder`. Does it address your concern?

Reply:
From my understanding:
1. The `logger` passed to `(f *Factory) CreateTraceExporter` and `(f *Factory) CreateMetricsExporter` is not used.
2. We created a new `logger` with the configured `logLevel` in `createLoggingExporter`. Let's call it `newLogger` to be differentiated from the previous one.
3. `newLogger` is passed to `NewTraceExporter` and `NewMetricsExporter`, and will be used to log the received traces/metrics. 

However, in `NewTraceExporter` and `NewMetricsExporter` we're still calling `newLogger.Debug` (or `newLogger.Info`). So the outputs will always be of level `debug` or `info`, and the previously configured log level is not respected. To make use of the log level, I think we need to update https://github.com/open-telemetry/opentelemetry-service/pull/257/files#diff-f20f48383dee213429a81b242322b371R37 to something like:

```go
logger.Log(exporterName, zap.Int("#spans", len(td.Spans)), logLevel)
```

(Syntax may not be correct but the idea is to use the log level instead of always logging at the same level.)

Reply:
It seems that there is no such logger.log method in the [zap package](https://godoc.org/go.uber.org/zap#Logger). In addition, I think it may be helpful to use different log level in the exporter. For example, later we may have

```go
logger.Info(exporterName, zap.Int("#spans", len(td.Spans)), logLevel)

// TODO: Add ability to record the received data
logger.Debug(exporterName, zap.Int("#spans", data)
```

Reply:
> In addition, I think it may be helpful to use different log level in the exporter.

That's right, but with the current approach I'm afraid [all logs will still be of level `debug`](https://github.com/open-telemetry/opentelemetry-service/blob/master/exporter/loggingexporter/logging_exporter.go#L33) even if you set it to other levels.

> It seems that there is no such logger.log method in the zap package. 

What about changing [NewTraceExporter](https://github.com/open-telemetry/opentelemetry-service/blob/master/exporter/loggingexporter/logging_exporter.go#L33) and [NewMetricsExporter](https://github.com/open-telemetry/opentelemetry-service/blob/master/exporter/loggingexporter/logging_exporter.go#L47) to:
```go
logger.Check(exporterName, logLevel).Write(zap.Int("#spans", len(td.Spans)))

logger.Check(exporterName, logLevel).Write(zap.Int("#metrics", len(md.Metrics)))
```

This way we can make sure the logs will be of the expected level.

Reply:
> > In addition, I think it may be helpful to use different log level in the exporter.
> 
> That's right, but with the current approach I'm afraid [all logs will still be of level `debug`](https://github.com/open-telemetry/opentelemetry-service/blob/master/exporter/loggingexporter/logging_exporter.go#L33) even if you set it to other levels.

My understanding is that for `logger.Debug(exporterName, zap.Int("#metrics", len(md.Metrics)))`, if the logger exporter's level is configured to `info` (as in the agent configuration file), the debug message will not be sent to stdout. So the log message is still being controlled in the configuration.

With that, the logging exporter will log messages in different levels by 

```go
// high priority log message
logger.Info(exporterName, zap.Int("#metrics", len(md.Metrics)))

// low priority log message to be added in the future
logger.Debug(exporterName, zap.Byte(data))
```

Please correct me if I am wrong (I have no objection to `using logger.Check()`). Thanks.

Reply:
I see, it seems Zap logger works in a different way than I thought. Thanks for the explanation.